shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/RewardUpdate.hs

```diff
@@ -160,7 +160,8 @@ emptyRewardUpdate =
 
 -- | To pulse the reward update, we need a snap shot of the EpochState particular to this computation
 data RewardSnapShot crypto = RewardSnapShot
-  { rewSnapshots :: !(SnapShots crypto),
+  { rewPstakeGo :: !(SnapShot crypto),
```

Comment:
The reward snap shot only needs the "go" snapshot to compute the rewards. The legacy code (pre-reward pulsing) was passed the whole SnapShots structure. So the Pulsing Rewards (NOT the stake distribution addressed in this PR) inherited passing the SnapShots. The problem with that is that the changes in SnapShots for stake distribution pulsing made it impossible to serialize the reward pulser without mutually recursive imports. This small change avoids that as the Reward Pulser now only contains the (go) snapShot and not the whole (mark,set,go) SnapShots.

Reply:
We only need the pstakeGo field. to compute rewards. It was a legacy artifact that the was inherited before we added pulsing to reward calculation. This simplifies things, and avoids a mutual recursive set of imports, because we no longer have to import SnapShots here, but just SnapShot.